File: INT1008 - Nhập môn lập trình/Tuần 5 - Hàm, khai báo định nghĩa. Truyền tham số cho hàm/Ngân hàng bài tập (Tuần 5)/Circle.cpp
#include <string>
using namespace std;

bool circle(string s) {
  int x = 0;
  int y = 0;
  for (int i = 0; i < int(s.length()); ++i) {
    if (s[i] == 'R') {
      ++y;
    } else if (s[i] == 'L') {
      --y;
    } else if (s[i] == 'U') {
      ++x;
    } else {
      --x;
    }
  }
  return (x == 0 && y == 0);
}

File: INT2204 - Lập trình hướng đối tượng/Tuần 5/Đa hình 1/Circle.java
public class Circle extends Shape {
  protected double radius;

  public Circle() {
    this.radius = 1;
  }

  public Circle(double radius) {
    this.radius = radius;
  }

  public Circle(double radius, String color, boolean filled) {
    super(color, filled);
    this.radius = radius;
  }

  public double getRadius() {
    return radius;
  }

  public void setRadius(double radius) {
    this.radius = radius;
  }

  public double getArea() {
    return 3.141592 * radius * radius;
  }

  public double getPerimeter() {
    return 2 * 3.141592 * radius;
  }

  public String toString() {
    return "Circle[radius=" + radius + ",color=" + super.getColor()
           + ",filled=" + super.isFilled() + "]";
  }
}
